File: mergeSortedArray.java
import java.util.*;

class mergeSortedArray {
    
    private int[] nums1;
    private int[] nums2;
    private int m;
    private int n;
    private int nums1Len;
    private int nums2Len;
    
    public mergeSortedArray(int[] nums1In, int nIn, int mIn, int[] nums2In)
    {
      nums1 = nums1In;
      nums2 = nums2In;
      n = nIn;
      m = mIn;
    }
    
    public void merge(){
        PriorityQueue<Integer> pQueue = new PriorityQueue<Integer>();
        for (int i = 0; i < n; i++){
            if (nums1[i] != 0){
            pQueue.add(nums1[i]);
        }
        }
        for (int i = 0; i < m; i++){
            if (nums1[i] != 0){
            pQueue.add(nums2[i]);
        }
        }
        for (int i = 0; i < n + m; i++){
            nums1[i] = pQueue.poll();
        }
        System.out.println(Arrays.toString(nums1));
        
    }
    
    
    public static void main(String[] args){
        int n = 3;
        int m = 3;
        int[] nums1 = {1,2,3,0,0,0};
        int[] nums2 = {2,5,6};
        mergeSortedArray mergeSorted = new mergeSortedArray(nums1, n, m, nums2);
        mergeSorted.merge();
        
    }
}

File: mergeSortedArray.py
def merge(nums1, m, nums2, n):
    lastPosition = m + n -1
    while m > 0 and n > 0:
        if nums1[m-1] > nums2[n-1]:
            nums1[lastPosition] = nums1[m-1]
            m -= 1
        else:
            nums1[lastPosition] = nums2[n-1]
            n -= 1
        lastPosition -= 1
    while n > 0:
        nums1[lastPosition] = nums2[n-1]
        n -= 1
        lastPosition -= 1
    return nums1





nums1= [2,2,3,0,0,0]
nums2 = [1,5,6]
m = 3
n = 3
print(merge(nums1, m, nums2, n))
